src/test/java/LoginFormTest.java

```diff
@@ -0,0 +1,115 @@
+import org.assertj.core.api.Assertions;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import pages.Homepage;
+import pages.LoginPage;
+
+
+import java.time.Duration;
+
+public class LoginFormTest {
+
+    private WebDriver driver;
+
+    @BeforeClass
+    public void setUp() {
+        System.setProperty("webdriver.chrome.driver", "src/main/resources/drivers/chromedriver.exe");
+        driver = new ChromeDriver();
+        driver.manage().window().maximize();
+        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(5));
+    }
+
+    @BeforeMethod
+    public void beforeMethod() {
+        driver.get("https://magento.softwaretestingboard.com/customer/account/login/");
+    }
+
+    @Test
+    public void loginPage_LoginWithValidCredentials_Success() {
+        //given
+        LoginPage loginpage = new Homepage(driver).openLoginPage();
+
+        //when
+        Homepage homepage = loginpage.loginWithEmailAndPassword("ala.k@wp.pl", "Password!1");
+
+        //then
+        Assertions.assertThat(homepage.getWelcomeText()).isEqualTo("Welcome, Ala Kowalska!");
+    }
+
+    @Test
+    public void loginPage_LoginWithValidEmailAndInvalidPassword_FailureToLogInWithErrorMessageIncorrectSignInDisplayed(){
+        //given
+        String validEmail = "ala.k@wp.pl";
+        String invalidPassword = "XXXXX";
+
+        //when
+        LoginPage loginPage = new LoginPage(driver).loginWithInvalidEmailAndPassword(validEmail, invalidPassword);
+
+        //then
+        Assertions.assertThat(driver.findElement(By.cssSelector(".message-error.error.message")).getText()).
+                isEqualTo("The account sign-in was incorrect or your account is disabled temporarily. Please wait and try again later.");
+    }
+    @Test
+    public void loginPage_LoginWithInvalidEmailAndInvalidPassword_FailureToLogInWithErrorMessageIncorrectSignInDisplayed(){
+        //given
+        String invalidEmail = "test@test.pl";
+        String invalidPassword = "XXXXX";
+
+        //when
+        LoginPage loginPage = new LoginPage(driver).loginWithInvalidEmailAndPassword(invalidEmail, invalidPassword);
+
+        //then
+        Assertions.assertThat(driver.findElement(By.cssSelector(".message-error.error.message")).getText()).
+                isEqualTo("The account sign-in was incorrect or your account is disabled temporarily. Please wait and try again later.");
+    }
+
+    @Test
+    public void loginPage_LoginWithInvalidEmailAndEmptyPassword_FailureToLogInWithErrorMessageThisIsRequiredFieldDisplayed(){
```

Comment:
`FailureWithCorrectFormErrorMessageDisplayed` instead of `FailureToLogInWithErrorMessageThisIsRequiredFieldDisplayed is enough for these and below tests.